umi_tools/umi_methods.py

```diff
@@ -30,13 +30,28 @@
     from _dedup_umi import edit_distance
 
 
-def get_umi(read, sep="_"):
-    try:
-        return read.qname.split(sep)[-1].encode('utf-8')
-    except IndexError:
-        raise ValueError("Could not extract UMI from read, please"
-                         "check UMI is encoded in the read name"
-                         "following the final '_' or run with --ignore-umi")
+def get_umi(read, sep="_", tag='RX', method='read_id'):
+    ''' extract the umi from either:
+    - read id using specified separator
+    - specified tag'''
+
+    if method == 'read_id':
```

Comment:
@IanSudbery I did profile and it seemed to make no difference whatsoever. I should probably check that again though.

I was originally going to set the get_umi method in the umi_methods.get_bundles function but as it's also used to generate stats in dedup. I presume we can set umi_getter in dedup as above and pass it to the umi_methods.get_bundles function?